### PR TITLE
Files on creation to be included in solution

### PR DESCRIPTION
**Pull request by annix8**

### CODE REVIEW

File: Sitefinity CLI/Commands/CommandBase.cs [hunk 1]
```diff
@@ -34,10 +34,12 @@ internal abstract class CommandBase
 
         protected bool IsSitefinityProject { get; set; } = true;
 
+        protected bool ShowAddFilesToProjectMessage { get; set; }
```
[the remark this answers is not in the document]
Comment (annix8) [reply Sitefinity CLI/Commands/CommandBase.cs]: Before, some of the commands always displayed "The file(s) should be added to the project manually.". Now, if CsProjModifier adds the file to the solution, the flag is raised and the commands do not display the message

File: Sitefinity CLI/CsProjModifier.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    projectElement.Add(itemGroupElement);
+                }
+
+                // sets the xmlns attr to be the one that is in the project element, so that no xmlns is added by default
+                compileElement = new XElement(projectElementXmlnsAttributeValue + Constants.CompileElem, new XAttribute(Constants.IncludeAttribute, filePath));
+                itemGroupElement.Add(compileElement);
+            }
+        }
+
+        public void RemoveFileFromCsProj(string filePath)
```
[the remark this answers is not in the document]
Comment (annix8) [reply Sitefinity CLI/CsProjModifier.cs]: removed

File: Sitefinity CLI/CsProjModifier.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Xml.Linq;
+
+[assembly: InternalsVisibleTo("Sitefinity CLI.Tests")]
+namespace Sitefinity_CLI
+{
+    internal class CsProjModifier
```
[the remark this answers is not in the document]
Comment (annix8) [reply Sitefinity CLI/CsProjModifier.cs]: done

**Review by Novkirishki — APPROVED** [review]

**Review by Gebov — APPROVED** [review]

File: Sitefinity CLI.Tests/AddTests.cs [hunk 4]
```diff
@@ -789,4 +789,4 @@ private List<float> GetAllTemplatesVersions(CultureInfo cultureInfo)
             return versions;
         }
     }
-}
+}
```
Comment (jordanilchev) [inline Sitefinity CLI.Tests/AddTests.cs]: ↩️

File: Sitefinity CLI/Commands/AddCustomWidgetCommand.cs [hunk 5]
```diff
@@ -51,6 +52,7 @@ public override int OnExecute(CommandLineApplication config)
             Directory.CreateDirectory(scriptsWidgetFolderPath);
 
             this.createdFiles = new List<string>();
+            var filesAddedToCsProjResult = new CsProjModifierResult();
```
Comment (Novkirishki) [inline Sitefinity CLI/Commands/AddCustomWidgetCommand.cs]: var filesAddedToCsProjResult = null;

File: Sitefinity CLI/FileAttributeModifier.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            FileAttributes attributes = GetFileAttributes(filePath);
+            attributes = RemoveAttribute(attributes, attributesToRemove);
+            SetFileAttributes(filePath, attributes);
+        }
+
+        public static void SetFileAttributes(string filePath, FileAttributes attributes)
+        {
+            File.SetAttributes(filePath, attributes);
+        }
```
Comment (Novkirishki) [inline Sitefinity CLI/FileAttributeModifier.cs]: new line

**Review by zzareva — APPROVED** [review]

**Pull request merged**